### PR TITLE
Implement RuleDesignOrdinal and DesignOrdinal classes

### PR DESCRIPTION
**Pull request by Puzzled-Face**

### CODE REVIEW

File: R/Design-class.R [hunk 1]
```diff
@@ -989,3 +989,208 @@ DesignGrouped <- function(model,
...
+
+#' `RuleDesignOrdinal`
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' [`RuleDesignOrdinal`] is the class for rule-based designs. The difference between
+#' this class and the [`DesignOrdinal`] class is that [`RuleDesignOrdinal`]
+#' does not contain `model`, `stopping` and `increments` slots.
+#'
+#' @slot next_best (`NextBest`)\cr how to find the next best dose.
```
Comment (danielinteractive) [inline R/Design-class.R:1005]: please adjust classes for the 3 relevant slots

File: R/Design-class.R [hunk 2]
```diff
@@ -989,3 +989,208 @@ DesignGrouped <- function(model,
...
+## class ----
+
+#' `DesignOrdinal`
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' [`DesignOrdinal`] is the class for rule-based ordinal designs. The difference
+#' between this class and its parent [`RuleDesignOrdinal`] class is that the
+#'  [`DesignOrdinal`] class contains additional `model`, `stopping` and
+#'  `increments` slots.
```
Comment (danielinteractive) [inline R/Design-class.R:1082]: also mention `pl_cohort_size` (or move it to the rule design class)

**Review by danielinteractive — COMMENTED** [review]
thanks @Puzzled-Face , see minor comments below

**Review by danielinteractive — COMMENTED** [review]
thanks @Puzzled-Face , see minor comments below

File: R/Design-class.R (hunk 2 above)
Comment (Puzzled-Face) [reply R/Design-class.R:1082]: I've kept `pl_cohort_size` in `DesignOrdinal` for consistency with `Design`.

**Review by danielinteractive — APPROVED** [review]

**Pull request merged**